blog/content/post/2024-07-25-odroid-h4-getting-started.md

```diff
@@ -0,0 +1,423 @@
+this little baby and see what we got...
+
+The platform came to me with small 3 boxes.
+
+![Unboxing](/img/odroid/odroid_boxes.jpg)
+
+The white one has a power supply. It also has a switchable plug, it is useful
+when you travel and may need to switch the plug to other countries' standards.
+
+![Unboxing](/img/odroid/odroid_psu.jpg)
+
+In the smaller dark box there is a small bag containing an eMMC module with
+Hardkernel signature. Typically you see eMMCs embedded on the board itself.
+However, the size constraints may have forced the designers to put it on a
+separate module, or to simply lower the board's price. Cool, we will figure
+out how to connect it later.
+
+![Unboxing](/img/odroid/odroid_emmc.jpg)
+
+And now the shining start of our show. The board... It came packed in a
+electrostatic-proof bag.
+
+![Unboxing](/img/odroid/odroid_board1.jpg)
+
+Underneath it we have two coin cell batteries. As the Hardkernel page says:
+`BIOS Backup Battery (All H series models include a backup battery by default)`
+and they do include it indeed.
+
+![Unboxing](/img/odroid/odroid_batteries.jpg)
+
+The board itself is so small that it fits into my hand.
+
+![Unboxing](/img/odroid/odroid_board2.jpg)
+
+And mine already has a RAM stick plugged in. According to board
+specifications, it has a single channel DDR5 memory. In single channel
+configuration it may reach a pretty frequency of up to 4800MHz (depending on
+the RAM stick as well of course).
+
+![Unboxing](/img/odroid/odroid_board3.jpg)
+
+## First launch
+
+Before we attempt any work, let's see if it even works... Things may happen in
+during the shipping, so ensure it boots properly and there are no signs of
+damage, before you start being rough with it. And believe me, we will (just a
+bit). Let's plug one of the coin cell batteries so that our BIOS setting will
+not get reset every boot (typically a CMOS battery reset will cause booting
+with default settings, like in last century PCs). Also connect the eMMC module
+to check if it is detected properly by the BIOS and OS.
+
+![Plugging eMMC](/img/odroid/odroid_emmc_plug.jpg)
+
+First boot may take a couple of minutes and include board resets, so be
+patient. The board reste may be observed by the blue LEDs going off. When they
+are on, board is booting/turned on:
+
+![ODROID power LEDs](/img/odroid/odroid_power_leds.jpg)
+
+After a while you should be presented with a BIOS setup (assuming no bootable
+mediums were connected):
+
+![ODROID BIOS setup](/img/odroid/odroid_setup.jpg)
+
+Apparently this board has two BIOS SPI flashes switchable by a jumper. That is
+great! We don't have to worry that much about recovery, we can always switch
+to the flash with working firmware in case something goes wrong. It will help
+a lot. Let's try to switch the flash and boot it to see if there is a valid
+firmware in the second flash chip. So let's swap the jumper near the DC jack
+as the Hardkernel page says:
```

Comment:
In the theory of operation (probably better for the Lab Assembly Guide or another document with procedures), we should describe how to use it during the development, debugging, and testing of firmware mods.